### PR TITLE
chore(deps): update dependency flutter to v3.24.1

### PR DESCRIPTION
**Pull request by Leptopoda**
fixes: #2391
replaces: #2399, #2404, #2415, #2400, #2406, #2397, #2398

Once reviewed, I'll sqash all commits together.
I had to disable custom_lint in our CI as it started analyzing files outside the package scope (mainly the symlinks to the pub cache with all our dependencies).

Now that we require dart 3.5 I'm looking into using dart workspaces.

### CODE REVIEW

**Comment by provokateurin** [comment]
> I had to disable custom_lint in our CI as it started analyzing files outside the package scope

This sounds bad, I assume it is an upstream bug that should be fixed before we continue here?

**Comment by provokateurin** [comment]
I would prefer to independently merge the package updates that are not necessary for updating the flutter version (should be rxdart, process_run and build_runner) and the pubspec.lock maintenance.

**Comment by provokateurin** [comment]
I assume you validated that the analyzer correctly works again in neon_framework?

**Comment by Leptopoda** [comment]
> This sounds bad, I assume it is an upstream bug that should be fixed before we continue here?

This is an old bug that we also had before. `custom_lint` basically scans for `*.dart` without caring where the file is located.
We might be able to work around it with a `melos exec "dart run custom_lint test && dart run custom_lint lib"` so it only scans in folders where we want to.

I assume you validated that the analyzer correctly works again in neon_framework?

I've tried this to my best and could not find any issues.

File: packages/neon_framework/pubspec.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ version: 1.0.0
 publish_to: 'none'
 
 environment:
-  sdk: ^3.0.0
+  sdk: ^3.5.0
```
Comment (provokateurin) [inline packages/neon_framework/pubspec.yaml:6]: Is this necessary?

File: packages/neon_framework/example/pubspec.lock
Comment (provokateurin) [inline packages/neon_framework/example/pubspec.lock:1]: To me it doesn't look like all the changes here are necessary.

**Review by provokateurin — COMMENTED** [review]
LGTM, but please fix the CI and also make sure to have a clean history of changes (so that we can properly revert if necessary).

**Pull request closed without merging**